lib/charms/kafka/v0/kafka_snap.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""## Overview
+
+`kafka_snap.py` provides a collection of common functions for managing snap installation and
+config parsing common to both the Kafka and ZooKeeper charms
+
+"""
+
+import subprocess
+import logging
+from typing import Dict, List
+
+from ops.model import StatusBase
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import snap
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "73d0f23286dd469596d358905406dcab"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 3
+
+
+class KafkaSnap:
+    def __init__(self) -> None:
+        self.default_config_path = "/snap/kafka/current/opt/kafka/config/"
```

Comment:
Are this two (`default_config_path` and `snap_config_path`) immutable?
If so, I suggest to set them as string literals instead of class attrs.

Reply:
This was sort-of a stop-gap for sharing functionality between K8s/VMs, as there won't be a snap on K8s.